src/Database/IdentifierInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.4.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+/**
+ * Represents a single identifier name in the database
+ */
+interface IdentifierInterface extends ExpressionInterface
```

Comment:
@thinkingmedia Check these:

* [IdentifierQuoter::quoteExpression()](https://github.com/cakephp/cakephp/blob/3.4.2/src/Database/IdentifierQuoter.php#L92)
* [IdentifierQuoter::_quoteIdentifierExpression()](https://github.com/cakephp/cakephp/blob/3.4.2/src/Database/IdentifierQuoter.php#L258)
* [QueryExpression::_calculateType()](https://github.com/cakephp/cakephp/blob/3.4.2/src/Database/Expression/QueryExpression.php#L793)

Identifier expressions are required for auto-quoting, AFAICT that's the whole point of having them in the first place, so at least for the sake of the argument I'd consider that as being part of the query builder.

So my point is that identifier expressions are a requirement for a specific functionality (which is also advertised in the Cookbook), and with a public API that serves such expressions, there should be a sufficient, realiable contract, even if the query builder doesn't directly expose a public API that would require specific interfaces or implementations, as failures to supply a proper implementation will result in auto-quoting not working, or even errors being thrown.